.drone.yml

```diff
@@ -74,6 +74,29 @@ steps:
           fi
         fi
 
+  # https://discourse.drone.io/t/how-to-exit-a-pipeline-early-without-failing/3951
+  # this step looks at the output of git diff --raw to determine whether any files
+  # which don't match the pattern '^docs/' or '.md$' were changed. if there are no
+  # changes to non-docs code, we skip the Teleport tests and exit early with a special
+  # Drone exit code to speed up iteration on docs (as milv is much quicker to run)
+  - name: Optionally skip tests
+    image: docker:git
+    commands:
+    - |
+        cd /go/src/github.com/gravitational/teleport
+        echo -e "\n---> git diff --raw ${DRONE_COMMIT}..origin/${DRONE_COMMIT_BRANCH:-master}\n"
+        git diff --raw ${DRONE_COMMIT}..origin/${DRONE_COMMIT_BRANCH:-master}
```

Comment:
```suggestion
```

Reply:
This line is here just to show the output of the command for reference so you know what is being evaluated by the line below. It's perhaps a bit hacky but it's useful to be able to see exactly what was changed in a PR (so we know whether we're running tests when we shouldn't be or, heaven forbid, vice versa)

Reply:
Here's some example output from the step so you can see what I mean:

```console
+ cd /go/src/github.com/gravitational/teleport\necho -e "\n---> git diff --raw 86dc48a8cbe39db9fc475a8200fc8811a8bc9655..origin/master\n"\ngit diff --raw 86dc48a8cbe39db9fc475a8200fc8811a8bc9655..origin/master\ngit diff --raw 86dc48a8cbe39db9fc475a8200fc8811a8bc9655..origin/master \| awk '{print $6}' \| grep -Ev '^docs/' \| grep -Ev '.md$' \| grep -v ^$ \| wc -l > /tmp/.change_count.txt\nexport CHANGE_COUNT=$(cat /tmp/.change_count.txt \| tr -d '\n')\necho -e "\n---> Changes to Teleport code detected: $CHANGE_COUNT"\nif [ $CHANGE_COUNT -gt 0 ]; then\n  echo "---> Teleport tests will run normally"\nelse\n  echo "---> Skipping Teleport tests and exiting early"\n  exit 78\nfi\necho ""\n

---> git diff --raw 86dc48a8cbe39db9fc475a8200fc8811a8bc9655..origin/master

100644 100644 c913486d 753f9551 M	.drone.yml

---> Changes to Teleport code detected: 1
---> Teleport tests will run normally
```